docs/external-helpers.md

```diff
@@ -0,0 +1,131 @@
+# Adding Externalized Helpers to the SDK
+
+## Introduction
+
+This guide explains how to add and manage externalized helper functions in the SDK, which interacts with ZAP (ZCL Advanced Platform). Initially, the SDK helpers were external to the ZAP repository, but due to maintenance and design concerns, they were moved into the ZAP repository. With the introduction of a new design pattern, SDK developers now have the flexibility to create their own externalized helpers and integrate them without requiring access to ZAP's source code.
+
+This design enables SDK developers to specify helper functions via the `gen-template.json` configuration file, which ZAP consumes during binary creation. The SDK can then call these helpers, passing in the necessary data, making them more flexible and reducing dependency on the internal ZAP codebase.
+
+## Background
+
+### Previous Approach
+
+In the earlier design, SDK helpers were external to ZAP. However, to function correctly, these helpers required access to ZAP’s source code, especially certain files within the ZAP repository. This created challenges:
+
+- When ZAP function names changed, corresponding updates had to be made across the SDK.
+- SDK developers were constrained by these internal dependencies, requiring explicit imports or "requires" of ZAP files.
+
+### New Design Pattern
+
+The new design pattern aims to free SDK developers from being tied to the internal structure of the ZAP repository, while still allowing the SDK to interact with ZAP's API. The new design works as follows:
```

Comment:
This should be "The design pattern", not "New design pattern".

Documentation should care about "what's there now", not "how did we get here".

Reply:
Fair enough, I'll remove it